src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIInteract.java

```diff
@@ -125,10 +127,14 @@ protected final boolean mineBlock(@NotNull final BlockPos blockToMine, @NotNull
         //add the drops to the citizen
         for (final ItemStack item : items)
         {
-            InventoryUtils.setStack(worker.getInventoryCitizen(), item);
+            ItemStack itemRemaining = ItemHandlerHelper.insertItemStacked(worker.getInventoryCitizen(), item, false);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'itemRemaining' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)